backend/src/main/java/com/rota/auth/AuthenticationUtils.java

```diff
@@ -1,51 +1,207 @@
 package com.rota.auth;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rota.api.StaffService;
 import com.rota.database.orm.staff.Role;
+import com.rota.database.orm.staff.Staff;
+import com.rota.exceptions.AuthenticationHttpException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.Charset;
 import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.json.JsonParseException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
 
 /**
- * Dummy class which represents user authentication.
+ * Utils for handling authorisation tokens and general authentication related methods.
  */
+@Service
 public class AuthenticationUtils {
+
+  @Value("${auth0.managementapi.client_secret}")
+  private String clientSecret;
+
+  @Value("${auth0.managementapi.client_id}")
+  private String clientId;
+
+  @Value("${auth0.managementapi.audience}")
+  private String audience;
+
+  @Autowired
+  StaffService staffService;
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final HttpClient httpClient = HttpClient.newBuilder()
+      .version(HttpClient.Version.HTTP_2)
+      .build();
+
   /**
-   * Checks if the provided token is valid.
-   * Only returns false if token is "false" (case insensitive).
-   * @param token Authentication token.
-   * @return Whether authentication was successful.
+   * TODO decide if it is definitely necessary to check with Auth0. Another method for that perhaps?
+   * Checks if the current token is valid locally and via Auth0.
+   *
+   * @return Whether the current token is valid.
    */
-  public static boolean validateToken(String token) {
-    // This will only return false if the token is "false" (case insensitive)
-    return !token.toLowerCase().equals("false");
+  public boolean validateToken(String token) {
+    HttpRequest request = HttpRequest.newBuilder()
+        .GET()
+        .uri(URI.create("https://rota-flex-101.eu.auth0.com/userinfo"))
```

Comment:
"https://rota-flex-101.eu.auth0.com/" is repeated in a few places, I suggest pulling it out as a constant.